Fix `test_agentd/test_agentd_state_config.py`

## Description

| Related issue |
|--|
| #1516 #1521 |

|Development branch|
|:--:|
|1588-agentd-state-config-test-fix|

When it comes to starting with the test fixture, there is some information that can help you:
```
Module: Agentd
Target: Agent   
OS: Windows
```
I add some information obtained while researched the logs that failed.

|  Type | Description  |
|--|--|
|Test Path | test_agentd/test_agentd_state_config.py
| Consistent | Yes
| Test Execution | 3/3 executions failed
| Cases Failed |   All (invalid values and one valid value)
| Summary | `test_agentd_state_config.py` is not starting the Wazuh service properly
| Error message |  " AssertionError: assert False is not False where False = check_if_process_is_running('wazuh-agentd')'"

In order to finish this issue the following tasks should be fulfilled:
- [x] Fails research.
- [x] Fix the test: The test needs to expect ValueError when it tries to start Wazuh with invalid values.
- [x] Full Green/Yellow in `test_agentd/test_agentd_state_config.py`. 

   The tests are fixed. However, they might fail because Wazuh is not showing error messages. There is an issue opened to fix this: https://github.com/wazuh/wazuh/issues/8746

- [x] Merge with `1521-full-green-agentd-fix`

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -35,6 +35,8 @@ with open(test_data_file) as f:
     test_cases = yaml.safe_load(f)
 
 # Variables
+wait_daemon_control = 1
+
 if sys.platform == 'win32':
     state_file_path = os.path.join(WAZUH_PATH, 'wazuh-agent.state')
     internal_options = os.path.join(WAZUH_PATH, 'internal_options.conf')
@@ -102,6 +104,9 @@ def test_agentd_state_config(test_case, set_local_internal_options):
             control_service('start')
     else:
         control_service('start', 'wazuh-agentd')
+        # Sleep enough time to Wazuh load agent.state_interval configuration and
+        # boot wazuh-agentd
+        time.sleep(wait_daemon_control) 
         assert (test_case['agentd_ends']
                     is not check_if_process_is_running('wazuh-agentd'))
     

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -79,7 +79,7 @@ def get_configuration(request):
 @pytest.mark.parametrize('test_case',
                          [test_case['test_case'] for test_case in test_cases],
                          ids=[test_case['name'] for test_case in test_cases])
-def test_agentd_state_config(test_case: list, set_local_internal_options):
+def test_agentd_state_config(test_case, set_local_internal_options):
 
     control_service('stop', 'wazuh-agentd')
 
@@ -95,11 +95,11 @@ def test_agentd_state_config(test_case: list, set_local_internal_options):
     if sys.platform == 'win32':
         if test_case['agentd_ends']:
             with pytest.raises(ValueError):
-                control_service('start', 'wazuh-agentd')
+                control_service('start')
             assert (test_case['agentd_ends']
                     is not check_if_process_is_running('wazuh-agentd'))
         else:
-            control_service('start', 'wazuh-agentd')
+            control_service('start')
     else:
         control_service('start', 'wazuh-agentd')
         assert (test_case['agentd_ends']

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -72,7 +72,7 @@ def get_configuration(request):
 @pytest.mark.parametrize('test_case',
                          [test_case['test_case'] for test_case in test_cases],
                          ids=[test_case['name'] for test_case in test_cases])
-def test_agentd_state_config(configure_environment, test_case: list):
+def test_agentd_state_config(test_case: list):
 
     control_service('stop', 'wazuh-agentd')
 

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -51,18 +51,25 @@ callbacks = {
 }
 
 
-# Functions
-def extra_configuration_before_yield():
-    change_internal_options('agent.debug', '2')
-
-
-def extra_configuration_after_yield():
-    # Set default values
-    change_internal_options('agent.debug', '0')
+# Fixture
+@pytest.fixture(scope='module')
+def set_local_internal_options():
+    """Set local internal options"""
+    if sys.platform == 'win32':
+        change_internal_options('windows.debug', '2')
+    else:
+        change_internal_options('agent.debug', '2')
+        
+    yield
+    
+    if sys.platform == 'win32':
+        change_internal_options('windows.debug', '0')
+    else:
+        change_internal_options('agent.debug', '0')
+        
     set_state_interval(5, internal_options)
 
 
-# Fixture
 @pytest.fixture(scope='module', params=configurations)
 def get_configuration(request):
     """Get configurations from the module."""
@@ -72,7 +79,7 @@ def get_configuration(request):
 @pytest.mark.parametrize('test_case',
                          [test_case['test_case'] for test_case in test_cases],
                          ids=[test_case['name'] for test_case in test_cases])
-def test_agentd_state_config(test_case: list):
+def test_agentd_state_config(test_case: list, set_local_internal_options):
 
     control_service('stop', 'wazuh-agentd')
 

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -85,8 +85,19 @@ def test_agentd_state_config(configure_environment, test_case: list):
     # Set state interval value according to test case specs
     set_state_interval(test_case['interval'], internal_options)
 
-    control_service('start', 'wazuh-agentd')
-
+    if sys.platform == 'win32':
+        if test_case['agentd_ends']:
+            with pytest.raises(ValueError):
+                control_service('start', 'wazuh-agentd')
+            assert (test_case['agentd_ends']
+                    is not check_if_process_is_running('wazuh-agentd'))
+        else:
+            control_service('start', 'wazuh-agentd')
+    else:
+        control_service('start', 'wazuh-agentd')
+        assert (test_case['agentd_ends']
+                    is not check_if_process_is_running('wazuh-agentd'))
+    
     # Check if test require checking state file existance
     if 'state_file_exist' in test_case:
         if test_case['state_file_exist']:
@@ -100,8 +111,3 @@ def test_agentd_state_config(configure_environment, test_case: list):
                             callback=callbacks.get(test_case['log_expect']),
                             error_message='Event not found')
     assert wazuh_log_monitor.result()
-
-    # Check if test require checking agentd status
-    if 'agentd_ends' in test_case:
-        assert (test_case['agentd_ends']
-                is not check_if_process_is_running('wazuh-agentd'))

--- a/tests/integration/test_agentd/test_agentd_state_config.py
+++ b/tests/integration/test_agentd/test_agentd_state_config.py
@@ -105,7 +105,7 @@ def test_agentd_state_config(test_case, set_local_internal_options):
         assert (test_case['agentd_ends']
                     is not check_if_process_is_running('wazuh-agentd'))
     
-    # Check if test require checking state file existance
+    # Check if the test requires checking state file existence
     if 'state_file_exist' in test_case:
         if test_case['state_file_exist']:
             # Wait until state file was dumped

